### ISSUE TITLE
A problem on zs.Concrete

### ISSUE THREAD

**Issue by rtz19970824** [thu-ml/zhusuan#84]
`
import tensorflow as tf
import zhusuan as zs

logits = tf.zeros([10, 10])
temparature = 1.
labels = zs.Concrete('y', logits=logits, temperature=temparature, n_samples=10, group_ndims=1)
prob = labels.log_prob(tf.one_hot(range(10), 10))
print (prob.shape)
`
seems only produce 1 tensor with shape ()

**Comment by thjashin** [comment]
@miskcoo Can you have a look?  btw, We don't recommend to use `zs.Concrete` outside `BayesianNet`. This is a historical problem and we will forbid this behavior in future. Use `zs.distributions.Concrete` instead.

**Comment by miskcoo** [comment]
@rtz19970824 I think this is right. Note that `zs.Concrete` is a multivariate distribution similar to `zs.OnehotCategorical`, so the input shape is like `batch_shape + input_shape`, where `input_shape` is the last dimension of `[10, 10]`, and `batch_shape` is the first dimension. Then you group last 1 dimension of `batch_shape`, `log_prob` will sum out this dim.

**Comment by rtz19970824** [comment]
@miskcoo I'm still a little confused. You mean the whole '[10, 10]' tensor is viewed as one sample so the output is only the probability of the whole sample? If I want to treat every single '[1, 10]' tensor  as logits of one distribution, how can I do?

**Comment by miskcoo** [comment]
Don't pass `group_ndims=1`, this parameter behaves like

`prob = tf.reduce_sum(prob, axis=0)`

**Comment by thjashin** [comment]
Does it work for you? @rtz19970824

**Comment by rtz19970824** [comment]
Yes, thanks.
